corporate/tests/test_stripe.py

```diff
@@ -3077,28 +3078,29 @@ def test_switch_now_free_trial_from_annual_to_monthly(self, *mocks: Mock) -> Non
                 self.assert_json_success(result)
                 plan.refresh_from_db()
                 self.assertEqual(plan.status, CustomerPlan.ENDED)
+                self.assertIsNone(plan.next_invoice_date)
 
-                plan = CustomerPlan.objects.get(
+                new_plan = CustomerPlan.objects.get(
                     customer=customer,
                     automanage_licenses=True,
                     price_per_license=800,
                     fixed_price=None,
                     discount=None,
                     billing_cycle_anchor=self.now,
                     billing_schedule=CustomerPlan.BILLING_SCHEDULE_MONTHLY,
-                    invoiced_through=None,
                     next_invoice_date=free_trial_end_date,
                     tier=CustomerPlan.TIER_CLOUD_STANDARD,
                     status=CustomerPlan.FREE_TRIAL,
                     charge_automatically=True,
                 )
-                LicenseLedger.objects.get(
-                    plan=plan,
+                ledger_entry = LicenseLedger.objects.get(
+                    plan=new_plan,
                     is_renewal=True,
                     event_time=self.now,
                     licenses=self.seat_count,
                     licenses_at_next_renewal=self.seat_count,
                 )
+                self.assertEqual(new_plan.invoiced_through, ledger_entry)
```

Comment:
I feel like we should extend this code to confirm what happens when invoicing runs, such that the test would have failed with the previous logic due to an extra invoice appearing.